C/services/south-plugin-interfaces/python/python_plugin_interface.cpp

```diff
@@ -154,145 +147,21 @@ void* PluginInterfaceResolveSymbol(const char *_sym)
-	pFunc = PyObject_GetAttrString(pModule, "plugin_info");
-	if (!pModule || !pFunc)
-		Logger::getLogger()->info("plugin_handle: plugin_info(): pModule=%p, pFunc=%p", pModule, pFunc);
-
-	if (!pFunc || !PyCallable_Check(pFunc))
-	{
-		// Failure
-		if (PyErr_Occurred())
-		{
-			logErrorMessage();
-		}
-
-		Logger::getLogger()->fatal("Cannot find method plugin_info in loaded python module");
-		Py_CLEAR(pFunc);
-
-		PyGILState_Release(state);
-		return NULL;
-	}
-	
-	// Call Python method passing an object
-	PyObject* pReturn = PyObject_CallFunction(pFunc, NULL);
-
-	Py_CLEAR(pFunc);
-
-	PLUGIN_INFORMATION *info = NULL;
-
-	// Handle returned data
-	if (!pReturn)
-	{
-		Logger::getLogger()->error("Called python script method plugin_info : error while getting result object");
-		logErrorMessage();
-		info = NULL;
-	}
-	else
-	{	
-		// Parse plugin information
-		info = Py2C_PluginInfo(pReturn);
-		
-		// bump interface version to atleast 2.x so that we are able to handle list of readings from python plugins in plugin_poll
-		if (info->interface[0]=='1' && info->interface[1]=='.')
-		{
-			Logger::getLogger()->info("plugin_handle: plugin_info(): Updating interface version from '%s' to '2.0.0' ", info->interface);
-			delete info->interface;
-			char *valStr = new char[6];
-			std::strcpy(valStr, "2.0.0");
-			info->interface = valStr;
-		}
-		
-		// Remove pReturn object
-		Py_CLEAR(pReturn);
-	}
-	if(info)
-		Logger::getLogger()->info("plugin_handle: plugin_info(): info={name=%s, version=%s, options=%d, type=%s, interface=%s, config=%s}", 
-					info->name, info->version, info->options, info->type, info->interface, info->config);
-	PyGILState_Release(state);
-	return info;
-}
-
-/**
- * Function to invoke 'plugin_init' function in python plugin
- */
-PLUGIN_HANDLE plugin_init_fn(ConfigCategory *config)
-{
-	PyObject* pFunc;
-	PyGILState_STATE state = PyGILState_Ensure();
-	
-	// Fetch required method in loaded object
-	pFunc = PyObject_GetAttrString(pModule, "plugin_init");
-	if (!pModule || !pFunc)
-		Logger::getLogger()->info("plugin_handle: plugin_init(): pModule=%p, pFunc=%p", pModule, pFunc);
-
-	if (!pFunc || !PyCallable_Check(pFunc))
-	{
-		// Failure
-		if (PyErr_Occurred())
-		{
-			logErrorMessage();
-		}
-
-		Logger::getLogger()->fatal("Cannot find method plugin_init in loaded python module");
-		Py_CLEAR(pFunc);
-
-		PyGILState_Release(state);
-		return NULL;
-	}
-
-	Logger::getLogger()->info("plugin_handle: plugin_init(): config->itemsToJSON()='%s'", config->itemsToJSON().c_str());
-	
-	// Call Python method passing an object
-	PyObject* pReturn = PyObject_CallFunction(pFunc, "s", config->itemsToJSON().c_str());
-
-	Py_CLEAR(pFunc);
-
-	// Handle returned data
-	if (!pReturn)
-	{
-		Logger::getLogger()->error("Called python script method plugin_init : error while getting result object");
-		logErrorMessage();
-
-		PyGILState_Release(state);
+		Logger::getLogger()->fatal("PluginInterfaceResolveSymbol returning NULL for sym=%s",
```

Comment:
Can we make this something that helps the user know what to do rather than something aimed at th eFogLAMP developer.